packages/loopring_v3/contracts/aux/access/ISubmitBlocksCallback.sol

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2017 Loopring Technology Limited.
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "../../core/iface/ExchangeData.sol";
+
+/// @title ISubmitBlocksCallback
+/// @author Brecht Devos - <brecht@loopring.org>
+abstract contract ISubmitBlocksCallback
```

Comment:
Fine with me, though the `DataReceiver` part may also be a bit confusing in that it's mostly used passively while it actively can do a lot of things.  `ISubmittedBlockReceiver`? I guess it's also more like a processor on the transaction level and not so much the full block level (because it also takes a transaction idx). `ISubmittedBlockProcessor`?

> IBeforeSubmissionBlockDataReceiver? maybe in the future we can also add IPostSubmissionBlockDataReceiver

If we need it we I think we may as well add a before/after to the same interface (with only small overhead if only one of the functions is needed).

Reply:
*XxxBlockReceiver* is fine.